### PR TITLE
A `view_manager` takes care of the different windows and switching betwwen them

### PR DESCRIPTION
**Pull request by TheoPannetier**
Fixes
- #553 
- #578 
- #582 
- #583 
- #584 

And completes Milestone 1!

### CODE REVIEW

File: game_view.cpp [hunk 1]
```diff
@@ -408,9 +409,8 @@ void test_game_view() //!OCLINT tests may be many
     }
 
     {
-    const game_view v;
-    // game has a member function called `get_n_ticks`, which returns zero upon construction
-    assert(v.get_game().get_n_ticks() == 0);
+        game_view v;    // game has a member function called `get_n_ticks`, which returns zero upon construction
```
Comment (Yangshen0325) [inline game_view.cpp:412]: I think we could keep "const" there. Is there a good reason why "const" is not used?

File: main.cpp [hunk 2]
```diff
@@ -161,67 +161,35 @@ int main(int argc, char **argv) //!OCLINT tests may be long
...
 
-  // Declare all views
-  options_view ov;
-  menu_view mv;
-  game_view gv(options);
-  assert(options == gv.get_options());
+  view_manager v{start_view, options};
+  v.exec();
+
+  return 0; // Game completes successfully
```
Comment (Yangshen0325) [inline main.cpp:192]: I enjoyed the simplified code there. Great!

File: view_manager.cpp [hunk 3]
```diff
@@ -1,18 +1,103 @@
...
+        assert(vw.get_next_view() != next_view);
+        vw.set_next_view(next_view);
+        assert(vw.get_next_view() == next_view);
+    }
+
+    /// (582) A view manager can run and exit
     {
-        view_manager{};
+       vw.set_next_view(view_mode::quit);
+       vw.exec();
```
Comment (Yangshen0325) [inline view_manager.cpp:97]: It seems good to re-use the same object multiple times, but it is unnecessary: It is better to have independent tests than tests that take a couple of seconds shorter. Please use a new/fresh view_manager every test. 🙂No need to optimize for speed yet 🙂 .

**Review by Yangshen0325 — CHANGES_REQUESTED** [review]
Great job! We think you could improve even more in the way as we described in the review. If you do so, we will happily accept.

File: game_view.cpp (hunk 1 above)
Comment (TheoPannetier) [reply game_view.cpp:412]: good point, fixed 👍

File: view_manager.cpp (hunk 3 above)
Comment (TheoPannetier) [reply view_manager.cpp:97]: gotcha, I reverted those changes to get back to independent tests.

Comment (richelbilderbeek) [reply view_manager.cpp:100]: Beautiful! Thanks!

**Review by richelbilderbeek — APPROVED** [review]
Good stuff! Thanks for keeping the tests independent :-)

**Pull request merged**